### PR TITLE
[hotfix][docs] Update IDE Setup guide

### PR DESCRIPTION
**Pull request by Airblader**
## What is the purpose of the change

This PR updates the IDE setup guide. I made some bigger structural changes such as reducing the heading levels, re-ordering sections and extracting a shared section for required plugins. Finally, I added a section for configuring the Python plugin in IntelliJ IDEA as discussed on the [mailing list](https://lists.apache.org/thread.html/r45e2e477dd57c6e9a27df9e192a115086ef29f9007fad4a7679c8541%40%3Cdev.flink.apache.org%3E).

I haven't copied over the change to the Chinese version of the file yet. Should I just overwrite the Chinese version? Or do I leave that intact for now?

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: docs/content/docs/flinkDev/ide_setup.md [hunk 1]
```diff
@@ -51,112 +51,43 @@ git clone https://github.com/apache/flink.git
...
-
-1. Start IntelliJ and choose New → Project from Existing Sources.
+1. Choose "New" → "Project from Existing Sources".
 2. Select the root folder of the cloned Flink repository.
 3. Choose "Import project from external model" and select "Maven".
 4. Leave the default options and successively click "Next" until you reach the SDK section.
 5. If there is no SDK listed, create one using the "+" sign on the top left.
    Select "JDK", choose the JDK home directory and click "OK".
-   Select the most suiting JDK version. NOTE: A good rule of thumb is to select 
+   Select the most suiting JDK version. NOTE: A good rule of thumb is to select
```
Comment (infoverload) [inline docs/content/docs/flinkDev/ide_setup.md:76]:
```suggestion
   Select the most suitable JDK version. NOTE: A good rule of thumb is to select
```

File: docs/content/docs/flinkDev/ide_setup.md [hunk 2]
```diff
@@ -176,76 +107,164 @@ Each file needs to include the Apache license as a header. This can be automated
...
 
-### Checkstyle For Python
-The Python code checkstyle of Apache Flink should create a flake8 external tool in the project. 
+1. Install flake8 for your Python interpreter using `pip install flake8`.
+2. In PyCharm go to "Preferences" → "Tools" → "External Tools".
+3. Select the "+" button to add a new external tool.
+4. Set "Name" to "flake8".
+5. Set "Description" to "Code Style Check".
+6. Set "Program" to the path of your Python interpreter, e.g. `/usr/bin/python`.
+7. Set "Arguments" to `-m flake8 \-\-config=tox.ini`.
```
Comment (dianfu) [inline docs/content/docs/flinkDev/ide_setup.md:264]:
```suggestion
7. Set "Arguments" to `-m flake8 --config=tox.ini`.
```

image

File: docs/content/docs/flinkDev/ide_setup.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ command line first (`mvn clean package -DskipTests`) as it might be your IDE
 that has a bug or is not properly set up.
 {{< /hint >}}
 
-## Preparation
```
[the remark this answers is not in the document]
Comment (Airblader) [reply docs/content/docs/flinkDev/ide_setup.md:42]: Good catch. That's a bummer, I guess it doesn't work the way I wanted it to. I'll have to increase the heading levels again.

**Comment by Airblader** [comment]
Thanks for your review @infoverload and @dianfu. I've updated the PR.

**Review by dianfu — APPROVED** [review]

**Pull request closed without merging**